R/fwrite.R

```diff
@@ -82,7 +82,8 @@ fwrite = function(x, file="", append=FALSE, quote="auto",
       return(invisible())
     }
   }
-  yaml = if (!yaml) "" else {
+  write_yaml <- yaml == TRUE || (is.list(yaml) && length(yaml) != 0)
```

Comment:
I guess this won't work in general:

```
list(1, 2, 3, 4, list(1, 4, 3, 2)) == TRUE
```

I think the LHS should be `isTRUE(yaml)` instead.

Please also use `0L` for `length` which is always `integer` 😬